### PR TITLE
Copy PrettyPrintJSON to core framework

### PR DESCRIPTION
**Pull request by oomichi**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

PrettyPrintJSON is most used e2emetrics function and that doesn't seem
specific for metrics. The implementation itself is generic, so it is
nice to move it to core framework for avoiding circular dependency.

Ref: https://github.com/kubernetes/kubernetes/issues/81427#issuecomment-525959908

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by fejta — APPROVED** [review]
/lgtm
/approve

**Comment by liggitt** [comment]
/hold

per [release-managers discussion](https://kubernetes.slack.com/archives/CJH2GBF7Y/p1567084773002200), holding low-priority, non-milestone-targeted PRs in order to free merge pool bandwidth for 1.16-targeted PRs

feel free to unhold once there are no [lgtm+approve+green 1.16-targeted PRs](https://github.com/kubernetes/kubernetes/pulls?utf8=✓&q=is%3Apr+is%3Aopen+label%3Algtm+label%3Aapproved+milestone%3Av1.16+status%3Asuccess) remaining.

**Review by fejta — APPROVED** [review]
/lgtm

**Pull request merged**
